cloud/passportauth/src/auth/PassportAuth.ts

```diff
@@ -103,13 +105,30 @@ export class PassportAuth implements Auth {
   }
 
   /**
+   * Checks if the current user has the role required to access a resource
+   *
+   * @param user - user data
+   * @param role - The required role to access a resource
+   * @returns Returns true/false if the user is authorized to access a resource
+   */
+  public hasResourceRole(user: any, role?: string) {
+    const userRoles = this.userService.getRoles(user);
+
+    if (role) {
+      return userRoles ? userRoles.indexOf(role) > -1 : false;
```

Comment:
@TommyJ1994 typescript will give you an error as indexOf can only take a param of type string.